src/portal-components/crc-researcher/ParticipantsBarPlot.tsx

```diff
@@ -0,0 +1,182 @@
+  let scheduledRows = data.filter(item => item['x'] === 'Scheduled')
+  let testedRows = data.filter(item => item['x'] === 'Tested')
+
+  const result: any[] = []
+  const defaultColors = [`black`, `#C4C4C4`]
+
+  let invitedCount = 0  
+  let scheduledCount = 0
+  let testedCount = 0
+
+  if (testedRows.length > 0) {
+    testedCount = parseInt(testedRows[0].count)
+  }
+
+  // note that if a participant was tested, then they were scheduled, and invited before
+  if (scheduledRows.length > 0) {
+    scheduledCount = parseInt(scheduledRows[0].count) + testedCount
+  }
+  // note that if a participant was scheduled, then they were invited before
+  if (invitedRows.length > 0) {
+    invitedCount = parseInt(invitedRows[0].count) + scheduledCount + testedCount
+  }
+
+  const x:string[] = ['Samples collected', 'Labs scheduled', 'Invitations sent']
+  const y1:number[] = [testedCount, scheduledCount, invitedCount]
+  const y2:number[] = [scheduledCount - testedCount, invitedCount - scheduledCount, totalCount - invitedCount]
+  const tooltips:string[] = [`${testedCount}/${scheduledCount}`, `${scheduledCount}/${invitedCount}`, `${invitedCount}/${totalCount}`]
+  result.push(getTraceItem(x, y1, 'Current', tooltips, defaultColors[0]))
+  result.push(getTraceItem(x, y2, 'Out of', tooltips, defaultColors[1]))
+  return result
+}
+
+function getLayout(
+  layoutConfig: Partial<PlotlyTyped.Layout>,
+  totalNumberOfResults: number
+): Partial<PlotlyTyped.Layout> {
+  const layout = _.cloneDeep(layoutConfig)
+  layout.xaxis = {
+    visible: false,
+  }
+  layout.title = {
+    text: `Total surveys received: ${totalNumberOfResults}`
+  }
+  layout.showlegend = false
+  layout.height = 240
+  return layout
+}
+
+export function fetchData(
+  token: string
+): Promise<RowSet> {
+  const sql = 'SELECT WorkflowState as "x", count(*) as "count" FROM syn22154087 GROUP BY WorkflowState'
+  
+  const queryRequest: QueryBundleRequest = {
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    partMask: SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+    entityId: 'syn22154087',
+    query: {
+      sql: sql,
+    },
+  }
+
+  return getFullQueryTableResults(queryRequest, token).then(
+    (data: QueryResultBundle) => {
+      return data.queryResult.queryResults
+    },
+  )
+}
+
+const ParticipantsBarPlot: FunctionComponent<ParticipantsBarPlotProps> = ({
+  token,
+  style = { width: '100%', height: '100%', margin: '30px 10px' },  
+}: ParticipantsBarPlotProps) => {
+// get plot data!
+const [isLoaded, setIsLoaded] = useState(false)
+const [barPlotQueryData, setBarPlotQueryData] = useState<GraphItem[]>([])
+useEffect(() => {
+  const barPlotData = fetchData(token!)
+  Promise.all([barPlotData])
+    .then(result => {
+      setBarPlotQueryData(resultToJson(result[0].headers, result[0].rows))
+      setIsLoaded(true)
+    })
+    .catch(err => {
+      throw err
+    })
+  return () => {}
+}, [token])
+
+  return (
+    <>
```

Comment:
Does the component need to send a message if the user isn't logged in?

Reply:
It's a good point, but I'd rather not tie this UI to each component.  
I don't want to see n components on the page all reporting that you need to sign in (lessons learned from synapse.org).